node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+            exceeding_balance,
+        );
+        let _ = payment_adjuster
+            .calculators
+            .into_iter()
+            .map(|calculator| calculator.calculate(&qualified_payable, &context))
+            .fold(0, |previous_result, current_result| {
+                let min = (current_result * 97) / 100;
+                let max = (current_result * 97) / 100;
+                assert_ne!(current_result, 0);
+                assert!(min <= previous_result || previous_result <= max);
+                current_result
+            });
+    }
+
+    type InputMatrixConfigurator = fn(
+        (QualifiedPayableAccount, QualifiedPayableAccount, SystemTime),
+    ) -> Vec<[(QualifiedPayableAccount, u128); 2]>;
+
+    #[test]
+    fn defaulted_calculators_react_on_correct_params() {
+        // When adding a test case for a new calculator, you need to make an array of inputs. Don't
+        // create brand-new accounts but clone the provided nominal accounts and modify them
+        // accordingly. Modify only those parameters that affect your calculator.
+        // It's recommended to orientate the modifications rather positively (additions), because
+        // there is a smaller chance you would run into some limit
+        let input_matrix: InputMatrixConfigurator =
+            |(nominal_account_1, nominal_account_2, _now)| {
+                vec![
+                    // First test case: BalanceCalculator
+                    {
+                        let mut account_1 = nominal_account_1;
+                        account_1.bare_account.balance_wei += 123_456_789;
+                        let mut account_2 = nominal_account_2;
+                        account_2.bare_account.balance_wei += 999_999_999;
+                        [(account_1, 8000001876543209), (account_2, 8000000999999999)]
+                    },
+                ]
+            };
+        // This is the value that is computed if the account stays unmodified. Same for both nominal
+        // accounts.
+        let current_nominal_weight = 8000000000000000;
+
+        test_calculators_reactivity(input_matrix, current_nominal_weight)
+    }
+
+    #[derive(Clone, Copy)]
+    struct TemplateComputedWeight {
+        common_weight: u128,
+    }
+
+    struct ExpectedWeightWithWallet {
+        wallet: Wallet,
+        weight: u128,
+    }
+
+    fn test_calculators_reactivity(
+        input_matrix_configurator: InputMatrixConfigurator,
+        nominal_weight: u128,
+    ) {
+        let calculators_count = PaymentAdjusterReal::default().calculators.len();
+        let now = SystemTime::now();
+        let cw_service_fee_balance_minor = gwei_to_wei::<u128, u64>(1_000_000);
+        let (template_accounts, template_computed_weight) =
+            prepare_nominal_data_before_loading_actual_test_input(
+                now,
+                cw_service_fee_balance_minor,
+            );
+        assert_eq!(template_computed_weight.common_weight, nominal_weight);
```

Comment:
There was an idea that you mentioned that you'd like to make `nominal_weight` a constant. I think it's a great idea. Let's do that.